test/e2e/framework.go

```diff
@@ -643,33 +649,54 @@ func (data *TestData) deleteTestNamespace(timeout time.Duration) error {
-	if err != nil || rc != 0 {
-		return fmt.Errorf("error when deploying Antrea; is %s available on the control-plane Node?", yamlFile)
+	deployYML := func(yml string) error {
+		rc, _, _, err := provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl apply %s -f %s", extraOptions, yml))
+		if err != nil || rc != 0 {
+			return fmt.Errorf("error when deploying Antrea; is %s available on the control-plane Node?", yml)
+		}
+		return nil
 	}
-	rc, stdout, stderr, err := provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl -n %s rollout status deploy/%s --timeout=%v", antreaNamespace, antreaDeployment, defaultTimeout))
-	if err != nil || rc != 0 {
-		return fmt.Errorf("error when waiting for antrea-controller rollout to complete - rc: %v - stdout: %v - stderr: %v - err: %v", rc, stdout, stderr, err)
+	rolloutAgent := func(ds string) error {
+		rc, stdout, stderr, err := provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl -n %s rollout status ds/%s --timeout=%v", antreaNamespace, ds, defaultTimeout))
+		if err != nil || rc != 0 {
+			return fmt.Errorf("error when waiting for antrea-agent (%s) rollout to complete - rc: %v - stdout: %v - stderr: %v - err: %v", ds, rc, stdout, stderr, err)
+		}
+		return nil
+	}
+
+	for _, yamlFile := range yamlFiles {
+		if err := deployYML(yamlFile); err != nil {
+			return err
+		}
 	}
 	if waitForAgentRollout {
-		rc, _, _, err = provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl -n %s rollout status ds/%s --timeout=%v", antreaNamespace, antreaDaemonSet, defaultTimeout))
-		if err != nil || rc != 0 {
-			return fmt.Errorf("error when waiting for antrea-agent rollout to complete - rc: %v - stdout: %v - stderr: %v - err: %v", rc, stdout, stderr, err)
+		if err := rolloutAgent(antreaDaemonSet); err != nil {
+			return err
 		}
+		if withWindowsAgent {
+			if err := rolloutAgent(antreaWindowsDaemonSet); err != nil {
+				return err
+			}
+		}
+	}
+
+	rc, stdout, stderr, err := provider.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl -n %s rollout status deploy/%s --timeout=%v", antreaNamespace, antreaDeployment, defaultTimeout))
+	if err != nil || rc != 0 {
+		return fmt.Errorf("error when waiting for antrea-controller rollout to complete - rc: %v - stdout: %v - stderr: %v - err: %v", rc, stdout, stderr, err)
 	}
 
 	return nil
 }
 
 // deployAntrea deploys Antrea with deploy options.
-func (data *TestData) deployAntrea(option deployAntreaOptions) error {
+func (data *TestData) deployAntrea(configuration deployAntreaConfiguration) error {
 	if testOptions.enableCoverage {
-		option = option.WithCoverage()
+		configuration = configuration.withCoverage()
 	}
-	return data.deployAntreaCommon(option.DeployYML(), "", true)
+	return data.deployAntreaCommon(configuration.getYML(), configuration.withWindowsAgents(), "", true)
```

Comment:
should `deployAntreaCommon` take the configuration as a parameter?

Reply:
Makes sense.